### PR TITLE
docs: ConfigurerAdapter to component-based security configuration

### PR DESCRIPTION
**Pull request by MarcinVaadin**
Information about new component-based security configuration instead of ConfigurerAdapter

Fixes: #1632

### CODE REVIEW

File: articles/security/enabling-security.adoc [hunk 1]
```diff
@@ -194,28 +199,12 @@ public class SecurityConfiguration
...
-    @Override
-    public void configure(WebSecurity web) throws Exception {
-        // Configure your static resources with public access here:
-        web.ignoring().antMatchers(
-                "/images/**"
-        );
-
-        // Delegating the ignoring configuration for Vaadin's
-        // related static resources to the super class:
-        super.configure(web); // <3>
```
Comment (mshabarov) [inline articles/security/enabling-security.adoc:208]: I propose to keep the `configure(WebSecurity web)` method in the code snippet to highlight the opportunity to override it and explain to devs for what it is. And you kept the point <3> below the code snippet.

File: articles/security/enabling-security.adoc [hunk 2]
```diff
@@ -245,6 +234,13 @@ Obviously, the implementation of the [methodname]`userDetailsService()` method i
 In a real-world application, you can change the Spring Security configuration to use an authentication provider for LDAP, JAAS, and other real-world sources.
 https://dzone.com/articles/spring-security-authentication[Read more about Spring Security authentication providers].
 
+.Component-based security configuration
+[NOTE]
+Spring Security 5.7.0-M2 deprecates the `WebSecurityConfigurerAdapter` and encourages users to move towards a component-based security configuration.
+Vaadin supports both approaches, although it is recommended to use component-based security configuration as in `SecurityConfiguration` example above.
+https://spring.io/blog/2022/02/21/spring-security-without-the-websecurityconfigureradapter[Read more about updating from WebSecurityConfigurerAdapter to component-based security configuration].
+`VaadinWebSecurityConfigurerAdapter` is still available for Vaadin 23.1 users but has been marked as deprecated and will be removed in the future.
```
Comment (mshabarov) [inline articles/security/enabling-security.adoc:242]:
```suggestion
`VaadinWebSecurityConfigurerAdapter` is still available for Vaadin 23.2 users but has been marked as deprecated and will be removed in the future.

```

File: articles/security/enabling-security.adoc [hunk 3]
```diff
@@ -245,6 +234,13 @@ Obviously, the implementation of the [methodname]`userDetailsService()` method i
 In a real-world application, you can change the Spring Security configuration to use an authentication provider for LDAP, JAAS, and other real-world sources.
 https://dzone.com/articles/spring-security-authentication[Read more about Spring Security authentication providers].
 
+.Component-based security configuration
+[NOTE]
+Spring Security 5.7.0-M2 deprecates the `WebSecurityConfigurerAdapter` and encourages users to move towards a component-based security configuration.
```
Comment (mshabarov) [inline articles/security/enabling-security.adoc:239]:
```suggestion
Spring Security 5.7.0 deprecates the `WebSecurityConfigurerAdapter` and encourages users to move towards a component-based security configuration.

```

**Review by mshabarov — CHANGES_REQUESTED** [review]

File: articles/security/enabling-security.adoc [hunk 4]
```diff
@@ -245,6 +234,13 @@ Obviously, the implementation of the [methodname]`userDetailsService()` method i
 In a real-world application, you can change the Spring Security configuration to use an authentication provider for LDAP, JAAS, and other real-world sources.
 https://dzone.com/articles/spring-security-authentication[Read more about Spring Security authentication providers].
 
+.Component-based security configuration
```
Comment (mshabarov) [inline articles/security/enabling-security.adoc:237]: I would also suggest to not use a note for this paragraph. Just place it before `This is how the view-based access control mechanism ...`.
Probably start the paragraph with ``VaadinWebSecurityConfigurerAdapter` is still available ...` is better, to make readers relax that they can still use the old class.

File: articles/security/enabling-security.adoc (hunk 1 above)
Comment (mshabarov) [reply articles/security/enabling-security.adoc:208]: For example, this user mentions migration guide for custom security implementation https://github.com/vaadin/flow/issues/14427#issue-1350521477 even.

Comment (MarcinVaadin) [reply articles/security/enabling-security.adoc:208]: TBH we don't use that method anymore as marking static resources using `ignoring()` is not proper way. Static resources have been moved to default `HttpSecurity` filter. I can leave that method empty:
```
/**
     * Allows customization of WebSecurity configuration.
     */
    @Override
    public void configure(WebSecurity web) throws Exception {
        // Vaadin by default does not customize WebSecurity.
        super.configure(web);
    }
```
But for me it pointless...

I don't fully understand your remark about point <3> - it is next to calling super.configure(HttpSecurity) as it was previously and there is matching explanation below.

File: articles/security/enabling-security.adoc (hunk 4 above)
Comment (MarcinVaadin) [reply articles/security/enabling-security.adoc:237]: I would reorganize those paragraphs like:
image

File: articles/security/enabling-security.adoc (hunk 1 above)
Comment (mshabarov) [reply articles/security/enabling-security.adoc:208]: Vaadin abstract class doesn't use it, indeed. But it doesn't mean that the extension classes wouldn't use `WebSecurity` API for their purposes. I would add this method as you showed in your comment above.

About <3> - my only concern was that you deleted the <3> next to the code line, but kept it in the description block.

Comment (MarcinVaadin) [reply articles/security/enabling-security.adoc:208]: There are 2 occurrences of <3>, there is one more in `configure(HttpSecurity)`

Comment (mshabarov) [reply articles/security/enabling-security.adoc:208]: Ah, okay, I didn't notice that.

**Review by mshabarov — CHANGES_REQUESTED** [review]
Looks good to me, but I set "request changes" to merge it after V23.2 GA is out.

**Review by mshabarov — APPROVED** [review]

**Pull request merged**